### PR TITLE
Refactor Cti::CallerId

### PR DESCRIPTION
**Pull request by corny**
- execute rebuild in a single transaction
- reduce queries
- DRY up code
- satisfy RuboCop

### CODE REVIEW

**Comment by corny** [comment]
Tests failed: `Mysql2::Error: Can't connect to local MySQL server through socket '/var/run/mysqld/mysqld.sock' (2)`
`rspec spec/models/cti/caller_id_spec.rb` works for me.

**Comment by monotek** [comment]
Seems to be a travis Bug:

https://github.com/travis-ci/travis-ci/issues/6961

**Comment by corny** [comment]
Ok, please test it locally.

File: app/models/cti/caller_id.rb [hunk 1]
```diff
@@ -23,10 +23,9 @@ def self.maybe_add(data)
         o_id:      data[:o_id],
         user_id:   data[:user_id],
       )
-      if record.new_record?
-        record.comment = data[:comment]
-        record.save!
-      end
+      return if !record.new_record?
+      record.comment = data[:comment]
+      record.save!
```
Comment (corny) [inline app/models/cti/caller_id.rb]: I prefer the previous version that always returns the record. If you want to keep you version better write `return record record if record.persisted?` or `return record unless record.new_record?`

Comment (thorsteneckel) [inline app/models/cti/caller_id.rb]: Thanks - done

**Comment by corny** [comment]
Looks fine for me. But I still prefer `unless` instead of `if !` ;-)

**Comment by corny** [comment]
Merge? :)

**Pull request merged**